src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/RoslynTestUtils.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.Generators.Test
+{
+    internal static class RoslynTestUtils
+    {
+        /// <summary>
+        /// Creates a canonical Roslyn project for testing.
+        /// </summary>
+        /// <param name="references">Assembly references to include in the project.</param>
+        /// <param name="includeBaseReferences">Whether to include references to the BCL assemblies.</param>
+        public static Project CreateTestProject(IEnumerable<Assembly>? references, bool includeBaseReferences = true)
+        {
+#pragma warning disable SA1009 // Closing parenthesis should be spaced correctly
+            var corelib = Assembly.GetAssembly(typeof(object))!.Location;
+            var runtimeDir = Path.GetDirectoryName(corelib)!;
+#pragma warning restore SA1009 // Closing parenthesis should be spaced correctly
+
+            var refs = new List<MetadataReference>();
+            if (includeBaseReferences)
+            {
+                refs.Add(MetadataReference.CreateFromFile(corelib));
+                refs.Add(MetadataReference.CreateFromFile(Path.Combine(runtimeDir, "netstandard.dll")));
+                refs.Add(MetadataReference.CreateFromFile(Path.Combine(runtimeDir, "System.Runtime.dll")));
+            }
+
+            if (references != null)
+            {
+                foreach (var r in references)
+                {
+                    refs.Add(MetadataReference.CreateFromFile(r.Location));
+                }
+            }
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+            return new AdhocWorkspace()
```

Comment:
```
System.TypeLoadException : Could not load type of field 'Microsoft.CodeAnalysis.SQLite.Interop.SafeSqliteBlobHandle:_wrapper' (0) due to: Could not load file or assembly 'SQLitePCLRaw.core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=1488e028ca7ab535' or one of its dependencies.
```
Stack trace:
```
   at System.RuntimeType.GetPropertiesByName(String name, BindingFlags bindingAttr, MemberListType listType, RuntimeType reflectedType) in /_/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:line 1901
   at System.RuntimeType.GetPropertyCandidates(String name, BindingFlags bindingAttr, Type[] types, Boolean allowPrefixLookup) in /_/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:line 624
   at System.RuntimeType.GetProperties(BindingFlags bindingAttr) in /_/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:line 729
   at System.Reflection.RuntimeReflectionExtensions.GetRuntimeProperties(Type type) in /_/src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeReflectionExtensions.cs:line 43
   at System.Composition.TypedParts.Discovery.TypeInspector.DiscoverPropertyExports(TypeInfo partType)+MoveNext()
   at System.Composition.TypedParts.Discovery.TypeInspector.DiscoverExports(TypeInfo partType)+MoveNext()
   at System.Composition.TypedParts.Discovery.TypeInspector.InspectTypeForPart(TypeInfo type, DiscoveredPart& part)
   at System.Composition.TypedParts.TypedPartExportDescriptorProvider..ctor(IEnumerable`1 types, AttributedModelProvider attributeContext)
   at System.Composition.Hosting.ContainerConfiguration.CreateContainer()
   at Microsoft.CodeAnalysis.Host.Mef.MefHostServices.Create(IEnumerable`1 assemblies)
   at Microsoft.CodeAnalysis.Host.Mef.MefHostServices.get_DefaultHost()
   at Microsoft.CodeAnalysis.AdhocWorkspace..ctor()
   at Microsoft.Extensions.Logging.Generators.Test.RoslynTestUtils.CreateTestProject(IEnumerable`1 references, Boolean includeBaseReferences) in /_/src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/RoslynTestUtils.cs:line 53
   at Microsoft.Extensions.Logging.Generators.Test.RoslynTestUtils.RunGenerator(ISourceGenerator generator, IEnumerable`1 references, IEnumerable`1 sources, AnalyzerConfigOptionsProvider optionsProvider, Boolean includeBaseReferences, CancellationToken cancellationToken) in /_/src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/RoslynTestUtils.cs:line 136
   at Microsoft.Extensions.Logging.Generators.Test.LoggerMessageGeneratorParserTests.RunGenerator(String code, Boolean wrap, Boolean inNamespace, Boolean includeBaseReferences, Boolean includeLoggingReferences, CancellationToken cancellationToken) in /_/src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorParserTests.cs:line 652
   at Microsoft.Extensions.Logging.Generators.Test.LoggerMessageGeneratorParserTests.NeedlessLogLevelInMessage() in /_/src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorParserTests.cs:line 136
--- End of stack trace from previous location ---
```

UPDATE:

the missing field `Microsoft.CodeAnalysis.SQLite.Interop.SafeSqliteBlobHandle:_wrapper` is in roslyn, and is of type `sqlite3_blob`:
https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/Workspaces/Core/Portable/Microsoft.CodeAnalysis.Workspaces.csproj#L25

And `SQLitePCLRaw` is referenced in [dotnet/roslyn/../Microsoft.CodeAnalysis.Workspaces.csproj#L25](https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/Workspaces/Core/Portable/Microsoft.CodeAnalysis.Workspaces.csproj#L25) as `PrivateAssets=all`

cc: @ericstj

Reply:
@tmat, does this make sense to you?  Why does roslyn have this dangling dependency and why isn't that a problem on CoreCLR?

Reply:
SQlite is an implementation detail only used conditionally if available.

Reply:
I repro'ed this with a one-liner (`new AdHocWorkspace();` on mono).  On mono it looks like reflection tries to initialize methods, which hit this dangling dependency.
https://github.com/dotnet/runtime/blob/bbb3dc4669b5c47ce7558026d68c201040191abd/src/mono/mono/metadata/icall.c#L4130

On CoreCLR and desktop it seems that `Type.GetProperties` do not try to JIT the methods/fields so they don't hit this.

@maryamariyan -- consider working around this by adding a reference to the missing assembly (just copy the PackageReference from roslyn repo).  I filed a bug: https://github.com/dotnet/runtime/issues/51148